test/unit/builder_disk_test.py

```diff
@@ -212,23 +220,23 @@ def test_setup_ix86(self, mock_machine):
         )
         assert disk_builder.arch == 'ix86'
 
-    @raises(KiwiInstallMediaError)
     @patch('kiwi.builder.disk.FileSystem')
-    @patch_open
+    @patch('builtins.open')
```

Comment:
Well, I can certainly do that, I even started doing it this way but I was facing a huge tests with all the code wrapped and intended inside the `with` clause, causing a massive diff and many breaks of the 80 lines max length, making the diff uglier, so that I opted for an in place replacement.

Is there any specific motivation to not patch the `builtins.open` as a decorator? Just willing to properly understand the refactor

Reply:
yes there is. The decorator patches the open for the complete test. However the test should be written in a way that only the code that issues an open is covered in a with() block. This allows to write tests and even use real open for e.g reading mock data that should be a return value for data read. Also if code that opens different files or files multiple times can be covered much better in a with() block

It's clear to me that the test file will change massive but that's part of the refactoring we agreed on. So I'm willing to accept this. Regarding the 80c line length I'm sure there will be a way to reformat that such that it fits in and probably become more readable.